pkg/kubelet/apis/kubeletconfig/v1beta1/types.go

```diff
@@ -44,6 +44,10 @@ const (
 type KubeletConfiguration struct {
 	metav1.TypeMeta `json:",inline"`
 
+	// certDirectory is the directory where the TLS certs are located (by
+	// default /var/run/kubernetes). If tlsCertFile and tlsPrivateKeyFile
```

Comment:
We can optionally use an empty default in the v1beta1 config, while maintaining the legacy default for flags.
I generally prefer to force explicit declarations of paths. 
@liggitt @mikedanese opinion?

Reply:
@mtaufen But with moving this flag into the config file, we can not maintain the legacy default for this flag.

Reply:
> But with moving this flag into the config file, we can not maintain the legacy default for this flag.

Not necessarily. We have independent control over the defaults for flag and struct.

> I'm less sure this path is the best one to start with in the kubelet config. do we have a good sense for what paths we plan to include in the config file?

I'll ask this another way. Do we have a good reason to keep this an independent path, or should we always place it under the kubelet's `--root-dir`?

Reply:
What would empty mean here? I definitely appreciate coherent config, but I'd save rejiggering of defaults for v1beta2. It's very easy to reason about a world where v1beta1 defaults are the canonical source for command line defaults.

Reply:
@liggitt what's the independent control? any specifications on that?
If we coherent with --root-dir, then the cert-dir would just be the postfix of the path.

Reply:
Or can we new another flag named cert-dir-postfix and deprecate the present cert-dir?

Reply:
> Or can we new another flag named cert-dir-postfix and deprecate the present cert-dir?

I doubt we want to do that. I'd like to hear from @kubernetes/sig-node-pr-reviews on what the desired structure of kubelet-owned directories is, and whether we want to consolidate some things under `--root-dir` or if it makes sense to continue having all of these separately configurable paths. Local filesystem paths are one of the most problematic thing to specify in a remotely-provided config shared by multiple nodes, so if we can minimize the number we have to deal with, that would be ideal

Reply:
@liggitt 
>Local filesystem paths are one of the most problematic thing to specify in a remotely-provided config shared by multiple nodes, so if we can minimize the number we have to deal with, that would be ideal

I agree with this, but I also think it's confusing to have some paths resolved against a root, while others are resolved against the config file location, in the same config version...I'd almost rather get rid of `--root-dir` for v1beta1 config files, and if/when we decide to consolidate under a root directory (say in v1beta2), we can change the behavior in a way that is consistent for all paths. (Not that I think it's this simple - there are probably classes of paths that make sense to deploy relative to the config file, and classes that make sense to deploy relative to a common root...)

One thing I would like to see in a v1beta2 is a logical grouping of config fields into substructures. At that point, it could make sense for related paths to share common roots, and it would be easy to group the roots into the same structures as the related paths.

@mikedanese Defaults already differ between v1beta1 and the command-line.